components/brave_wallet/browser/cardano/cardano_transaction_serializer.cc

```diff
@@ -36,16 +77,158 @@ cbor::Value::ArrayValue CardanoTransactionSerializer::SerializeInputs(
   cbor::Value::ArrayValue result;
   for (const auto& input : tx.inputs()) {
     cbor::Value::ArrayValue input_value;
+
     input_value.emplace_back(input.utxo_outpoint.txid);
     // TODO(https://github.com/brave/brave-browser/issues/45278): upstream a fix
     // for cbor::Value to support uint64_t
     input_value.emplace_back(static_cast<int64_t>(input.utxo_outpoint.index));
+
     result.emplace_back(std::move(input_value));
   }
 
   return result;
 }
 
+std::optional<
+    std::vector<CardanoTransactionSerializer::RestoredTransactionInput>>
+CardanoTransactionSerializer::DeserializeInputs(
+    const cbor::Value::ArrayValue& data) {
+  std::vector<RestoredTransactionInput> tx_inputs;
+  for (const auto& value : data) {
+    if (!value.is_array() || value.GetArray().size() != 2) {
+      return std::nullopt;
+    }
+
+    if (!value.GetArray()[0].is_bytestring()) {
+      return std::nullopt;
+    }
+
+    if (value.GetArray()[0].GetBytestring().size() != kCardanoTxHashSize) {
+      return std::nullopt;
+    }
+
+    if (!value.GetArray()[1].is_integer()) {
+      return std::nullopt;
+    }
+
+    RestoredTransactionInput tx_input;
+    base::span(tx_input.tx_hash)
+        .copy_from_nonoverlapping(value.GetArray()[0].GetBytestring());
+    tx_input.index = static_cast<uint64_t>(value.GetArray()[1].GetInteger());
```

Comment:
should not be static_cast'ed to uint64_t, should be checked if positive and casted to uint32